Dev category database

bvkeersop opened the pull request:
Added category and createCategory endpoint + exceptions

bvkeersop committed:
Added some exceptions

bvkeersop committed:
Removed exceptions that don't really do anything.

bvkeersop committed:
Exception added when updating vehicle but licenseplate already exists

bvkeersop committed:
commit, forgot to base branch of previous branch...

bvkeersop committed:
Merge branch 'dev-exceptionHandling' into dev-categoryDatabase

Merged exception branch into this one, and added category control and exceptions

bvkeersop committed:
Added createCategory endpoint and exceptions

bvkeersop committed:
change persistence.xml to update

juleskreutzer commented on src/main/java/rest/CategoryResource.java:
```diff
@@ -0,0 +1,38 @@
+	CategoryService categoryService;
+
+	@POST
+	@Path("/create")
```
Waarschijnlijk handiger om ipv POST /create een POST te hebben op / (zonder create).
Is een GET om alle categorieën op te halen ook nodig?

bvkeersop committed:
Added exceptions for empty fields

Did not check on licensplate yet, ask productowner if this has to be implemented.

bvkeersop committed:
Implemented code review comments and some other stuff.

- Also implemented correct endpoint names for all resources (no need to use /get, if the header contains get it works).
- Implemented exception handling for empty fields, and for updating a vehicle
